File: DKS-SPA/src/app/core/_models/s_f406i.ts
import { Pagination } from "./pagination";
import { utilityConfig } from "../utility/utility-config";

export class SF406i extends Pagination {

  stockNo: string;
  materialNo: string;
  type: number;
  /**
   *default set of searching parameters
   */
  constructor() {
    super();
    this.isPaging = false; //開分頁
    this.stockNo = "";
    this.materialNo = "";
    this.type = 0;
  }

  public setPagination(pagination: Pagination) {
    this.currentPage = pagination.currentPage;
    this.itemsPerPage = pagination.itemsPerPage;
    this.totalItems = pagination.totalItems;
    this.totalPages = pagination.totalPages;
  }
}

File: DKS-API/DTOs/SF406i.cs
using System;
using DKS_API.Helpers;

namespace DKS_API.DTOs
{
    public class SF406i : PaginationParams
    {
        public string StockNo { get; set; }
        public string  MaterialNo { get; set; }
        public int Type { get; set; }
    }
}